src/governance.h

```diff
@@ -147,11 +163,19 @@ class CGovernanceManager
     void UpdatedBlockTip(const CBlockIndex *pindex);
     int64_t GetLastDiffTime() {return nTimeLastDiff;}
     void UpdateLastDiffTime(int64_t nTimeIn) {nTimeLastDiff=nTimeIn;}
+
+    int GetCachedBlockHeight() { return nCachedBlockHeight; }
```

Comment:
The purpose of this is to prevent deadlocks.  If we don't cache the block height we need to lock cs_main to get it.  It then becomes more difficult to ensure that the lock order between cs_main and governance.cs (which protects the governance data structures) is always the same.